sklego/preprocessing/identitytransformer.py

```diff
@@ -9,16 +9,27 @@ class IdentityTransformer(BaseEstimator, TransformerMixin):
     The identity transformer returns what it is fed. Does not apply anything useful.
     The reason for having it is because you can build more expressive pipelines.
     """
+    def __init__(
+        self,
+        check_X: bool = True
+    ):
+        self.check_X = check_X
 
     def fit(self, X, y=None):
         """'Fits' the estimator."""
-        X = check_array(X, copy=True, estimator=self)
+        if self.check_X:
+            X = check_array(X, copy=True, estimator=self)
+        else:
+            X = X.copy()
```

Comment:
Why does `X` need to be copied here?

Reply:
`check_array` function also copies input data. The idea was even if array is not checked, data is copied for consistency. Should I remove it?